sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala

```diff
@@ -85,6 +90,43 @@ class HiveExternalCatalogVersionsSuite extends SparkSubmitTestUtils {
     new File(tmpDataDir, name).getCanonicalPath
   }
 
+  private def getFileFromUrl(urlString: String, targetDir: String, filename: String): Boolean = {
+    val conf = new SparkConf
+    val securityManager = new SecurityManager(conf)
+    val hadoopConf = new Configuration
+
+    val outDir = new File(targetDir)
+    if (!outDir.exists()) {
+      outDir.mkdirs()
+    }
+
+    try {
+      val result = Utils.doFetchFile(urlString, outDir, filename, conf, securityManager, hadoopConf)
+      result.exists()
+    } catch {
+      case ex: Exception => logError("Could not get file from url " + urlString + ": "
+        + ex.getMessage)
+        false
+    }
+  }
+
+  private def getStringFromUrl(urlString: String, encoding: String = "UTF-8"): String = {
+    val outDir = Files.createTempDirectory("string-")
+    val filename = "string-out.txt"
+
+    if (!getFileFromUrl(urlString, outDir.toString, filename)) {
+      throw new java.io.IOException("Could not get string from url " + urlString)
+    }
+
+    val outputFile = new File(outDir.toString + File.separator + filename)
```

Comment:
I think you want to use the constructor that takes File, String rather than construct the path. MInor thing.
I'd also say use the newer Path API but may not be what Scala APIs use.